c/enc/find_match_length.h

```diff
@@ -32,7 +34,14 @@ static BROTLI_INLINE size_t FindMatchLengthWithLimit(const uint8_t* s1,
     } else {
       uint64_t x = BROTLI_UNALIGNED_LOAD64LE(s2) ^
           BROTLI_UNALIGNED_LOAD64LE(s1 + matched);
+#if BROTLI_GNUC_HAS_BUILTIN(__builtin_ctzll, 3, 4, 0) || \
+    BROTLI_INTEL_VERSION_CHECK(16, 0, 0)
       size_t matching_bits = (size_t)__builtin_ctzll(x);
+#else /* BROTLI_MSVC_INTRINSICS */
+      uint32_t first_one;
+      _BitScanForward64(&first_one, x);
```

Comment:
`_BitScanForward64` only supported with 64-bit compiler (`defined(_M_AMD64)`). I think you should use `_BitScanForward`

Something like that:
```
  size_t matching_bits;
  if (_BitScanForward(&first_one, x))
    matching_bits = (size_t)first_one;
  else if (_BitScanForward(&first_one, (x >> 32)))
    matching_bits = (size_t)(where + 32);
```

Reply:
The whole function body is wrapped in `#if BROTLI_64_BITS`, so that `#else` would never happen.  I've refactored the change and used `_tzcnt_u64`, so hopefully it's a more appealing PR now.

Reply:
Not all processors support `_tzcnt_u64` (`BMI1 is available in AMD's Jaguar,[4] Piledriver[5] and newer processors, and in Intel's Haswell[6] and newer processors. `)

I think better revert to _BitScanForward

Reply:
See the earlier comments on this PR.  TZCNT decodes as BSF on non-BMI1 processors
https://stackoverflow.com/questions/43880227/why-does-tzcnt-work-for-my-sandy-bridge-processor